sips/sip-021/sip-021-nakamoto.md

```diff
@@ -0,0 +1,1085 @@
+
+
+<em>Figure 3: Tenures N+1 and N+2 are empty.  Stackers recover from this by
+requiring the next non-empty tenure to include the empty tenures'
+<code>TenureChange</code> transactions, to prove that these tenures were in fact
+empty and no blocks were signed for them.  Because the tenures are empty, the
+inventory bitmap would mark them as 0s -- there is no block data to fetch.</em>
+
+To recover from this, Stackers would still produce `TenureChange` transactions
+for the empty tenures, and require the start block of the first non-empty tenure
+to include them.  In other words, Stackers _always_ produce a `TenureChange`
+transaction for each tenure; it just might not be included by the tenure's
+sortition-selected miner.
+
+### Stacker Turnover
+
+A miner tenure change happens every Bitcoin block.  In addition, there are
+Stacker tenure changes, which happen once every 2100 Bitcoin blocks (one reward
+cycle).  In a Stacker tenure change, a new set of Stackers are selected by the
+PoX anchor block (see SIP-007).
+
+Because Stacks no longer forks, the PoX anchor block is always known 100 Bitcoin
+blocks prior to the start of the next reward cycle:  it is simply the the start
+block of the 2000th tenure in the current reward cycle.
+
+The PoX anchor block identifies the next Stackers.  They have 100 Bitcoin blocks
+to prepare for signing Stacks blocks.  Within this amount of time, the new
+Stackers would complete a WSTS DKG for signing blocks (among other things, such
+as the sBTC peg wallet hand-off).  The PoX contract will require Stackers to
+register their block-signing keys when they stack or delegate-stack STX, so the
+entire network knows enough information to validate their WSTS Schnorr
+signatures on blocks.
+
+### PoX Failure
+
+In the event that PoX does not activate, the chain halts.  If there are no
+Stackers, then block production cannot happen.
+
+### Changes to PoX
+
+To support tenure changes, this proposal calls for a new PoX contract, `.pox-4`.
+The `.pox-4` contract would be altered over the current PoX contract (`.pox-3`)
+to meet the following requirements:
+
+* Stackers register a WSTS signing key when they call `stack-stx` and optionally
+  `delegate-stx`.
+    * In `delegate-stx`, Stackers may instead elect for the delegate to supply
+      the signing key.  In this case, the delegate carries out the Stacker's
+responsibility for signing blocks.
+* The stacking minimum is abolished.  If there is at least one stacker that is
+  able to claim one reward slot, then PoX activates.
+* The `.pox-4` contract will expose each reward cycle's full reward set,
+  including the signing keys for each stacker, via a public function.
+Internally, the Stacks node will call a _private_ function to load the reward
+set into the `.pox-4` data space after it identifies the PoX anchor block.  This
+is required for other dependent SIPs, such as the sBTC SIP.
+
```

Comment:
```suggestion
* Solo stacking is removed to simplify the stacking protocol. The functions `stack-stx`, `stack-extend`, `stack-increase` are removed.
* Decrease of stacked amount is added. The function `delegate-stack-decrease` is added.
```

Reply:
Solo stacking is removed? does it mean someone who wants to stack on their own uses the pool-operater route instead (a pool with a single delegate)?

Reply:
Negative. Solo stacking is not removed afaik.

Reply:
This is my proposal. It would simplify things.

Reply:
The relevant issue on stacks-core is https://github.com/stacks-network/stacks-core/issues/4182

Solo stacking is used less and less and for the solo stackers security shouldn't be used at all. Hot wallet as pool operator makes more sense and is adopted by whales and institutional holders already.

Reply:
Can you clarify "the solo stackers security shouldn't be used at all?" A bit unclear to parse through that.

Reply:
The security in general is increased the less the wallet holding large amounts is used.
Currently, solo stackers can use desktop Leather wallet with limited features or browser extensions with higher risk of being tricked into bad txs.
Using a solo pool would reduce the risk and increase the number of features that can be used.

Reply:
Maybe delegate-stx-decrease might be out of scope for v1

Reply:
The reason we didn't try to add a `-decrease` method for stacking and delegate-stacking is because it significantly complicates the auto-unlock logic.  Without careful design it could introduce a DoS vector, whereby someone could stack and then decrease to 1 uSTX over many, many iterations in order to blow up the number of unlocks each node would have to process.  Can you post a more detailed design proposal for `-decrease`?

Reply:
Update: I removed the suggestion to add `-decrease`